### PR TITLE
Fixes bug that allowed schedules in the same quarter to have the same name

### LINKED ISSUE #9: BUG: You can make Personal Schedules in same quarter with duplicate names thorugh edit
### Have you updated the title?

- [X] I have updated the issue's title.

### Summary

Personal schedules are intended to have unique names within a given quarter; this is so that they can be distinguished in the modal for adding courses to a schedule.

The code for creating personal schedules prevents you from making a new personal schedule in a given quarter with a certain name.  If you try it, you'll get this error message:

image

However, you can work around this using edit, since the edit feature doesn't do the same check.

The bug fix is to make edit behave like add in this respect.

### Expected Behavior

If you edit a personal schedule you should not be able to change the name to one that already exists in that quarter for that user.

### Current / Observed Behavior

You can edit a schedule to change its name to one that already exists in the quarter.

### Steps to Reproduce

1. Create a personal schedule for a given quarter (lets say W22) and call it W22a
2. Create a second schedule for the same quarter, and try to also call it W22a.  You'll get the error message (this is correct behavior so far). So change the name to W22b
3. Now edit the W22b schedule.  Change the name to W22a, and click save
4. The schedule will be saved.  (What we want is the same error message as with create.)

### Browser / Platform Info

Browser doesn't matter here.

### PR DESCRIPTION
**Pull request by bsaiju**
Closes #9 

Disallows personal schedules that are in the same quarter to be edited to have the same name by adding extra logic in the Personal Schedule Edit Page and Controller.

# Before Fix

https://github.com/user-attachments/assets/6c198499-1204-4dba-9cdc-a0ac44df42c7

# After Fix

https://github.com/user-attachments/assets/971127e6-4347-4ac1-b7a3-0c500c088eed

# Dokku Link
[https://courses-bsaiju.dokku-03.cs.ucsb.edu/](https://courses-bsaiju.dokku-03.cs.ucsb.edu/)

### CODE REVIEW

**Review by pballani — APPROVED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java [hunk 1]
```diff
@@ -185,6 +193,14 @@ public PersonalSchedule putScheduleById_admin(
...
     personalschedule.setQuarter(incomingSchedule.getQuarter());
 
+    Optional<PersonalSchedule> existCheck =
+        personalscheduleRepository.findByUserAndNameAndQuarter(
+            incomingSchedule.getUser(), incomingSchedule.getName(), incomingSchedule.getQuarter());
+    if (existCheck.isPresent()) {
+      throw new IllegalArgumentException(
+          "A personal schedule with that name already exists in that quarter");
+    }
+
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java:203]: There's a bug in this logic.  See the comment on the PR.

**Review by pconrad — CHANGES_REQUESTED** [review]
This implementation contains a bug.  See the comments on the PR.

**Comment by pconrad** [comment]
Closed, due to no response from team on Wednesday 11/17/2024:

image

**Pull request closed without merging**